aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -32,116 +31,191 @@ Tensor& fill_(Tensor& self, const Tensor& value) {
+}
+
+Tensor& clamp_(Tensor& self, Scalar value1, Scalar value2) {
+  return self._clamp_(value1, value2);
+}
+
+Tensor& clamp_out(Tensor& result, const Tensor& self, Scalar value1, Scalar value2) {
+  result.resize_(self.sizes());
+  result.copy_(self);
+  return result._clamp_(value1, value2);
+}
+
+Tensor clamp_max(const Tensor& self, Scalar value) {
+  Tensor result = self.type().tensor();
+  return native::clamp_max_out(result, self, value);
+}
+
+Tensor& clamp_max_(Tensor& self, Scalar value) {
+  return self._clamp_max_(value);
+}
+
+Tensor& clamp_max_out(Tensor& result, const Tensor& self, Scalar value1) {
+  result.resize_(self.sizes());
+  result.copy_(self);
+  return result._clamp_max_(value1);
+}
+
+Tensor clamp_min(const Tensor& self, Scalar value) {
+  Tensor result = self.type().tensor();
+  return native::clamp_min_out(result, self, value);
+}
+
+Tensor& clamp_min_(Tensor& self, Scalar value) {
+  return self._clamp_min_(value);
+}
+
+Tensor& clamp_min_out(Tensor& result, const Tensor& self, Scalar value1) {
+  result.resize_(self.sizes());
+  result.copy_(self);
+  return result._clamp_min_(value1);
+}
+
+Tensor frac(const Tensor& self) {
+  Tensor result = self.type().tensor();
+  return frac_out(result, self);
+}
+
+Tensor& frac_(Tensor& self) {
+  return frac_out(self, self);
+}
+
+Tensor& _frac_out_cpu(Tensor& result, const Tensor& self) {
+  result.resize_(self.sizes());
+  return at::_th_frac_out(result, self);
+}
+
+Tensor erfinv(const Tensor& self) {
+  Tensor result = self.type().tensor();
+  return erfinv_out(result, self);
+}
+
+Tensor& _erfinv__cpu(Tensor& self) {
+  return _erfinv_out_cpu(self, self);
+}
+
+Tensor& _erfinv_out_cpu(Tensor& result, const Tensor& self) {
+  result.resize_(self.sizes());
+  return at::_erfinv_out(result, self);
+}
+
+Tensor sigmoid(const Tensor& self) {
+  Tensor result = self.type().tensor();
+  return sigmoid_out(result, self);
+}
+
+Tensor& sigmoid_(Tensor& self) {
+  return sigmoid_out(self, self);
+}
+
+Tensor& _sigmoid_out_cpu(Tensor& result, const Tensor& self) {
+  result.resize_(self.sizes());
+  return at::_th_sigmoid_out(result, self);
+}
+
+Tensor clone(const Tensor& self) {
+  return self.type()._th_clone(self);
+}
+
+Tensor _contiguous_cpu(const Tensor& self) {
+  return self.type()._th_contiguous(self);
+}
+
+Tensor neg(const Tensor& self) {
+  Tensor result = self.type().tensor();
+  return neg_out(result, self);
+}
+
+Tensor& neg_(Tensor& self) {
+  return neg_out(self, self);
+}
+
+Tensor& _neg_out_cpu(Tensor& result, const Tensor& self) {
+  result.resize_(self.sizes());
+  return at::_th_neg_out(result, self);
+}
+
+Tensor reciprocal(const Tensor& self) {
+  Tensor result = self.type().tensor();
+  return reciprocal_out(result, self);
+}
+
+Tensor& reciprocal_(Tensor& self) {
+  return reciprocal_out(self, self);
+}
+
+Tensor& _reciprocal_out_cpu(Tensor& result, const Tensor& self) {
+  result.resize_(self.sizes());
+  return at::_th_reciprocal_out(result, self);
+}
```

Comment:
Those looks quite repetitive again. Why can't we use the macros below to implement them? If they are not vectorized, why are the moving to ATen, and why can't we have an unvectorized macro for them?

Reply:
I'm just writing out all the boilerplate code and will then vectorize them one by one.